javascript/ql/src/semmle/javascript/frameworks/ClientRequests.qll

```diff
@@ -0,0 +1,194 @@
+/**
+ * Provides classes for modelling the client-side of a URL request.
+ *
+ * Subclass `ClientRequest` to refine the behavior of the analysis on existing client requests.
+ * Subclass `CustomClientRequest` to introduce new kinds of client requests.
+ */
+
+import javascript
+
+/**
+ * A call that performs a request to a URL.
+ */
+class CustomClientRequest extends DataFlow::InvokeNode {
```

Comment:
I believe this was intended to be abstract?

Reply:
ouch. Good catch.

Reply:
Would this have been caught by the tests? If not, is it possible to add a regression test?

Reply:
Nope. The unit tests used the `getUrl` predicate, so the mistake was invisible. Furthermore, the unit test sources did not have a negative case.
I have amended the fixup commit with some regression tests.